docs/tutorial.md

```diff
@@ -0,0 +1,369 @@
+---
+title: Quarkus Operator Tutorial
+linkTitle: Tutorial
+weight: 30
+description: An in-depth walkthough of building and running a Quarkus-based operator.
+---
+
+## Prerequisites
+
+- Java through the [installation guide](https://java.com/en/download/help/download_options.html).
+- User authorized with `cluster-admin` permissions.
+- Maven installation [installation guide](https://maven.apache.org/install.html)
+
+## Overview
+
+We will create a sample project to let you know how it works and this sample will:
+
+- Create a Memcached Deployment if it doesn't exist
+- Ensure that the Deployment size is the same as specified by the Memcached CR spec
+- Update the Memcached CR status using the status writer with the names of the CR's pods
+
+## Create a new project
+
+Use the CLI to create a new memcached-operator project:
+
+```sh
+mkdir memcached-quarkus-operator
+cd memcached-quarkus-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --plugins quarkus --domain example.com --project-name memcached-quarkus-operator
+```
+
+#### A note on dependency management
+
+`operator-sdk init` generates `pom.xml` file. This file contains all the dependencies required to run the operator.
+
+### MemcachedQuarkusOperator
+
+The main program of the operator `JavaOpOperator.java` initializes and runs the operator. The operator uses java-operator-sdk which is similar to the Go lang version of controller-runtime.
+
+The code below will initialize and define the informers/watches for your operator.
+
+```
+  @Override
+  public int run(String... args) throws Exception {
+    operator.start();
+
+    Quarkus.waitForExit();
+    return 0;
+  }
+```
+
+## Create a new API and Controller
+
+Create a new Custom Resource Definition (CRD) API with group `cache` version `v1alpha1` and Kind `Memcached`. The plugin, still in its alpha state, will output debug messages which are normal.
```

Comment:
This should match the items you are passing in to the command below.

```suggestion
Create a new Custom Resource Definition (CRD) API with group `cache` version `v1` and Kind `Memcached`. The plugin, still in its alpha state, will output debug messages which are normal.
```

Reply:
Modified.